pkg/plugins/agent_config.go

```diff
@@ -48,6 +49,16 @@ func (ac *AgentConfigPlugin) Run() {
 	value := reflect.ValueOf(ac.config)
 	for i := 0; i < value.NumField(); i++ {
 		name := value.Type().Field(i).Name
+
+		// serialise only public fields
+		if len(name) <= 0 {
+			continue
+		}
+		firstLetter := rune(name[0])
+		if unicode.IsLower(firstLetter) {
```

Comment:
shouldn't we filter them by the "public" annotation?
https://github.com/newrelic/infrastructure-agent/blob/master/pkg/config/config.go#L64
In case that a config option needs to be public in the code but we don't wan't to send the value

Reply:
I think that "public" annotation belongs to the past agent closed source history, where we wanted to know whether a config option was meant to be publicly announced.

Reply:
BTW do we still use this "public" anno for something?

Reply:
Used for discriminating which fields should be logged and now we'll be using it for agent config inventory plugin.